net/socket/net_sendfile.c

```diff
@@ -130,8 +130,8 @@ ssize_t psock_sendfile(FAR struct socket *psock, FAR struct file *infile,
   if (psock == NULL || psock->s_conn == NULL)
     {
       nerr("ERROR: Invalid socket\n");
-      _SO_SETERRNO(psock, EBADF);
-      return ERROR;
+      psock->s_error = EBADF;
```

Comment:
`psock_sendfile` documentation must be updated to reflect the current implementation. The __Return Value__ section still states that on failure it returns -1 and sets `errno`.

Reply:
Done.